examples/gpt-2/main.cpp

```diff
@@ -299,14 +330,25 @@ bool gpt2_model_load(const std::string & fname, gpt2_model & model, gpt_vocab &
         const size_t memory_size = ggml_nbytes(model.memory_k) + ggml_nbytes(model.memory_v);
 
         printf("%s: memory size = %8.2f MB, n_mem = %d\n", __func__, memory_size/1024.0/1024.0, n_mem);
+
+        // allocate buffer and tensors
+        model.buffer_kv = ggml_backend_alloc_buffer(model.backend, memory_size + 256);
+        ggml_allocr * alloc = ggml_allocr_new_from_buffer(model.buffer_kv);
+        ggml_allocr_alloc(alloc, model.memory_k);
+        ggml_allocr_alloc(alloc, model.memory_v);
+        ggml_allocr_free(alloc);
```

Comment:
What is the purpose of this `ggml_allocr` here?

Reply:
To allocate the tensors in the backend buffer. It is overkill, and we probably should add a simpler allocator that cannot free tensors, but since this is only done during initialization it doesn't seem like a big deal to use the big allocator.